src/workerd/api/crypto-impl.c++

```diff
@@ -210,4 +211,24 @@ ZeroOnFree::~ZeroOnFree() noexcept(false) {
   OPENSSL_cleanse(inner.begin(), inner.size());
 }
 
+void checkPbkdfLimits(jsg::Lock& js, size_t iterations) {
+  KJ_IF_SOME(workerIsolate, Worker::Isolate::from(js)) {
+    auto& limits = workerIsolate.getLimitEnforcer();
+    KJ_IF_SOME(max, limits.checkPbkdfIterations(js, iterations)) {
+      JSG_FAIL_REQUIRE(DOMNotSupportedError,
+          kj::str("Pbkdf2 failed: iteration counts above ", max ," are not supported (requested ",
+                  iterations, ")."));
+    }
+  } else {
+    // If the jsg::Lock embedder data has not been set then we're running in an
+    // environment that does not yet pass along the necessary information for the
```

Comment:
What sort of environment would that be? Can we make it so the new embedder data pointer is _never_ null?

Reply:
Well, the current internal repo for one.  Along with potentially anyone else theoretically embedding workerd given that the lock itself is created in the embedder-specific API implementation (where the embedder data would be set).

As with other changes I want to do this in two steps so that we can land the workerd change, then separately update the internal repo and make the internal change. Then once the internal repo is updated I can come back and make it so the embedder pointer is never null.

Reply:
However, avoiding this two-step dance would likely make a good argument for making this an embedder data pointer on the v8::Isolate rather than the lock since that can be done separately from actually creating the lock itself.

Reply:
I don't see any benefit to doing this as a three-step process, just do the lockstep change. It's less work, reduces the complexity of the code, and ensures that we don't leave stuff unfinished leading to bugs down the road.

Reply:
Refactored to use the v8::Isolate embedder data pointer.